In [1]:
import pandas as pd 
import numpy as np 

In [150]:
bruggen = pd.read_csv('bruggen.csv')
groen = pd.read_csv('groen_bruggen.csv', converters={'idnr': '{:0>8}'.format})

In [151]:
groen.head()

provincie      gewest  vnr      idnr oud-brugnr   gemeente (deelgemeente)  \
0  Antwerpen  Vlaanderen  220  00010751         O5    SCHOTEN            NaN   
1  Antwerpen  Vlaanderen  221  00010791         O9     BRECHT         BRECHT   
2  Antwerpen  Vlaanderen  222  00010741         B4    SCHOTEN            NaN   
3  Antwerpen  Vlaanderen  223  00130101         B5  WOMMELGEM      WOMMELGEM   
4  Antwerpen  Vlaanderen  224  00130131         B8      RANST            NaN   

                                  brugnaam weg\nboven weg\nonder  ...  \
0  BRUG-O5-JAGERSDREEF-OVER-DUWVAARTKANAAL       A001       W203  ...   
1                      Brug O9 Kapelstraat       A001       G000  ...   
2                      BRUG-B4-ELSHOUTBAAN       N121       A001  ...   
3               Brug B5 - Wijnegemsteenweg       G000       A013  ...   
4                      BRUG-B8-VAARTSTRAAT       G000       A013  ...   

   longitude probleem oorzaak werken_gepland start_werken bezocht opmerkingen  \
0        NaN      NaN     NaN            NaN          NaN     NaN         NaN   
1  4.615.664      NaN     NaN            NaN          NaN     NaN         NaN   
2  4.511.460      NaN     NaN            NaN          NaN     NaN         NaN   
3  4.527.529      NaN     NaN            NaN          NaN     NaN         NaN   
4  4.570.469      NaN     NaN            NaN          NaN     NaN         NaN   

   beelden  hoogproblematisch  lijst_sinds  
0      NaN                nee          NaN  
1      NaN                nee          NaN  
2      NaN                nee          NaN  
3      NaN                 ja       2007.0  
4      NaN                 ja          NaN  

[5 rows x 31 columns]

In [152]:
groen.gemeente = groen.gemeente.str.title()

In [57]:
gewest = bruggen[bruggen.lijst_sinds.notna()]

In [153]:
groen_gewest = groen[groen.lijst_sinds.notna()]

In [154]:
groen.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2803 entries, 0 to 2802
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   provincie                2803 non-null   object 
 1   gewest                   2803 non-null   object 
 2   vnr                      2803 non-null   int64  
 3   idnr                     2803 non-null   object 
 4   oud-brugnr               1526 non-null   object 
 5   gemeente                 2803 non-null   object 
 6   (deelgemeente)           1225 non-null   object 
 7   brugnaam                 2803 non-null   object 
 8   weg
boven                2795 non-null   object 
 9   weg
onder                2792 non-null   object 
 10  beheerder                2803 non-null   object 
 11  beheer                   2803 non-null   object 
 12  toestandcode             2764 non-null   object 
 13  huidige toestand         2764 non-null   object 
 14  LPK                     

In [60]:
merged = gewest.merge(groen_gewest, on=['provincie', 'gemeente', 'beheerder', 'lijst_sinds'], how='outer')

In [61]:
merged.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 68 entries, 0 to 67
Data columns (total 43 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   gemeente                 68 non-null     object 
 1   gewest_x                 68 non-null     object 
 2   provincie                68 non-null     object 
 3   brugnaam_x               68 non-null     object 
 4   latitude_x               68 non-null     float64
 5   longitude_x              68 non-null     float64
 6   probleem_x               68 non-null     object 
 7   oorzaak_x                68 non-null     object 
 8   werken_gepland_x         68 non-null     object 
 9   start_werken_x           68 non-null     object 
 10  bezocht_x                68 non-null     object 
 11  opmerkingen_x            35 non-null     object 
 12  beelden_x                34 non-null     object 
 13  beheerder                68 non-null     object 
 14  hoogproblematisch_x      68 

In [62]:
merged.to_csv('merged_orig.csv', sep=';', index=False)

In [213]:
prio = pd.read_csv('merged.csv', sep=';', converters={'idnr': '{:0>8}'.format})

In [214]:
prio.head()

gemeente      gewest       provincie    latitude  longitude  \
0  Wommelgem  Vlaanderen       Antwerpen  51.212.246  4.527.518   
1     Deurne  Vlaanderen       Antwerpen  51.210.954  4.464.007   
2     Deurne  Vlaanderen       Antwerpen  51.214.129  4.459.276   
3   Arendonk  Vlaanderen       Antwerpen  51.338.964  5.104.604   
4   Zaventem  Vlaanderen  Vlaams Brabant  50.880.841  4.460.796   

                                            probleem      oorzaak  \
0  Vallende brokstukken. Afgesloten voor alle gem...     brokkelt   
1               Jaarlijkse opvolging en waterpassing    technisch   
2  3-maandelijke inspectie, 6-maandelijkse waterp...    technisch   
3                         Renovatieplannen in opmaak     onbekend   
4             Moet vernieuwd worden bij verbreden R0  groter plan   

  werken_gepland start_werken bezocht  ... Unnamed: 33 Unnamed: 34  \
0            Nee     onbekend     Nee  ...         NaN         NaN   
1             Ja         2021     Nee  ...         NaN         NaN   
2             Ja         2022     Nee  ...         NaN         NaN   
3             Ja         2020     Nee  ...         NaN         NaN   
4             Ja         2022     Nee  ...         NaN         NaN   

  Unnamed: 35 Unnamed: 36  Unnamed: 37  Unnamed: 38 Unnamed: 39 Unnamed: 40  \
0         NaN         NaN          NaN          NaN         NaN         NaN   
1         NaN         NaN          NaN          NaN         NaN         NaN   
2         NaN         NaN          NaN          NaN         NaN         NaN   
3         NaN         NaN          NaN          NaN         NaN         NaN   
4         NaN         NaN          NaN          NaN         NaN         NaN   

  Unnamed: 41 Unnamed: 42  
0         NaN         NaN  
1         NaN         NaN  
2         NaN         NaN  
3         NaN         NaN  
4         NaN         NaN  

[5 rows x 43 columns]

In [215]:
andere = bruggen[bruggen.lijst_sinds.isna()]

In [216]:
groen_andere = groen[groen.lijst_sinds.isna()]

In [217]:
concat = pd.concat([prio, groen_andere], axis=0, ignore_index=True)

In [218]:
concat = concat.drop(columns=['Unnamed: 31', 'Unnamed: 32', 'Unnamed: 33', 'Unnamed: 34', 'Unnamed: 35', 'Unnamed: 36', 'Unnamed: 37', 'Unnamed: 38', 'Unnamed: 39', 'Unnamed: 40', 'Unnamed: 41', 'Unnamed: 42'])

In [219]:
new = pd.concat([concat, andere], axis=0, ignore_index=True)

In [220]:
new.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2902 entries, 0 to 2901
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   gemeente                 2902 non-null   object 
 1   gewest                   2902 non-null   object 
 2   provincie                2902 non-null   object 
 3   latitude                 492 non-null    object 
 4   longitude                491 non-null    object 
 5   probleem                 131 non-null    object 
 6   oorzaak                  131 non-null    object 
 7   werken_gepland           138 non-null    object 
 8   start_werken             138 non-null    object 
 9   bezocht                  78 non-null     object 
 10  opmerkingen              29 non-null     object 
 11  beelden                  20 non-null     object 
 12  beheerder                2902 non-null   object 
 13  hoogproblematisch        2902 non-null   object 
 14  lijst_sinds             

In [221]:
new.idnr = new.idnr.astype(str).str.rstrip('.0')

In [232]:
new = new.rename(columns={'weg\nonder': 'weg_onder', 'weg\nboven': 'weg_boven', 'voorlaatste\nA-inspectie': 'voorlaatste_a_inspectie', 'recentste\nA-inspectie': 'recentste_a_inspectie', 'diff': 'difference'})

In [223]:
new.lijst_sinds = new.lijst_sinds.astype(str).str.rstrip('.0')

In [224]:
new.vnr = new.vnr.astype(str).str.strip('.0')

In [228]:
new.prev_year = new.prev_year.astype(str).str.rstrip('.0')

In [230]:
new.last_year = new.last_year.astype(str).str.rstrip('.0')

In [233]:
new.difference  = new.difference.astype(str).str.rstrip('.0')

In [246]:
new = new[['idnr', 'gemeente', '(deelgemeente)', 'provincie', 'gewest', 'brugnaam', 'probleem', 'oorzaak', 'werken_gepland', 'start_werken', 'bezocht', 'opmerkingen', 'beelden', 'beheerder', 'beheer', 'toestandcode', 'LPK', 'prev_year', 'last_year', 'difference', 'hoogproblematisch', 'lijst_sinds']]

In [247]:
len(new[(new.duplicated('idnr')) & (new.idnr != 'nan') & (new.toestandcode != 'Toestand 1')].index)

130

In [251]:
new.to_csv('new.csv', index=False, sep=';')

In [3]:
final = pd.read_csv('new_dups.csv', sep=';', converters={'idnr': '{:0>8}'.format})

In [4]:
final = final.replace('00000nan', 'nan')

In [5]:
len(final[(final.duplicated('idnr')) & (final.idnr != 'nan') & (final.toestandcode != 'Toestand 1')].index)

130

In [6]:
final.lijst_sinds = final.lijst_sinds.astype(str).str.rstrip('.0')
final.prev_year = final.prev_year.astype(str).str.rstrip('.0')
final.last_year = final.last_year.astype(str).str.rstrip('.0')
final.difference  = final.difference.astype(str).str.rstrip('.0')

In [7]:
final['(deelgemeente)'] = final['(deelgemeente)'].str.title()

In [8]:
df = final.loc[(final.idnr == 'nan') | (~final.idnr.duplicated())]

In [302]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2749 entries, 0 to 2901
Data columns (total 22 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   idnr               2749 non-null   object
 1   gemeente           2749 non-null   object
 2   (deelgemeente)     1162 non-null   object
 3   provincie          2749 non-null   object
 4   gewest             2749 non-null   object
 5   brugnaam           2749 non-null   object
 6   probleem           131 non-null    object
 7   oorzaak            131 non-null    object
 8   werken_gepland     138 non-null    object
 9   start_werken       138 non-null    object
 10  bezocht            78 non-null     object
 11  opmerkingen        29 non-null     object
 12  beelden            20 non-null     object
 13  beheerder          2749 non-null   object
 14  beheer             2650 non-null   object
 15  toestandcode       2611 non-null   object
 16  LPK                43 non-null     object


In [9]:
df = df[df.idnr != '00100401']

In [345]:
df.beheerder.unique()

array(['AWV', 'AWV/VWW', 'Vlaamse Waterweg', 'DMOW', 'MDK', 'AWV/DMOW',
       'Gemeente', 'Brussels Gewest', 'Infrabel'], dtype=object)

In [352]:
len(df[(df.gewest == 'Vlaanderen') & (df.beheerder != 'Gemeente') & (df.beheerder != 'Brussels Gewest') & (df.beheerder != 'Infrabel')].index)

2649

In [353]:
len(df[(df.hoogproblematisch == 'ja') & (df.gewest == 'Vlaanderen') & (df.beheerder != 'Gemeente') & (df.beheerder != 'Brussels Gewest') & (df.beheerder != 'Infrabel')])

593

In [444]:
df.to_csv('bruggen_volledig.csv', sep=';', index=False)

# Analyse

## Hoeveel hoogproblematische bruggen zijn er in totaal en wat is de verhouding tov de totale hoeveelheid bruggen?
Deze vraag is eenvoudig te beantwoorden voor de beheerders waarvoor we precies weten hoeveel bruggen zij beheren. Die informatie hebben we voor het Vlaams Gewest, Infrabel en het Brussels Gewest. Van de gemeentes weten we niet hoeveel bruggen zij precies beheren. We kunnen dus ook niet zeggen hoeveel bruggen in totaal per provincie hoogproblematisch zijn, of hoeveel bruggen totaal in provincie gerenoveerd moeten worden. Dat is een belangrijk onderscheid. Deze totalen staan in de spreadsheet onder het tabblad 'aantallen'. 

## Hoeveel hoogproblematische bruggen zijn er tov bruggen die zeker gerenoveerd moeten worden?
Hierboven heb ik eerst alle datasets opgeschoond en terug op elkaar geplakt en daarna nog alle duplicaten eruit gehaald (bruggen waar zowel een boven- als onderbeheerder aanwezig is, waardoor de brug er twee keer in staat, maar het gaat effectief om dezelfde brug.) Ook heb ik alle bruggen met Toestand 1, onbekende toestand of geen beoordeling eruit gehaald: we kijken nu alleen naar de bruggen waarvan we weten dat ze gerenoveerd moeten worden. Dit omdat we anders alleen van het Vlaams Gewest daadwerkelijk alle bruggen in onze lijst hebben.

In [559]:
frame = df[(df.toestandcode != 'Toestand 1') & (df.toestandcode != 'Toestand t') & (df.toestandcode != 'Geen beoor')]

In [560]:
frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1767 entries, 0 to 2901
Data columns (total 22 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   idnr               1767 non-null   object
 1   gemeente           1767 non-null   object
 2   (deelgemeente)     771 non-null    object
 3   provincie          1767 non-null   object
 4   gewest             1767 non-null   object
 5   brugnaam           1767 non-null   object
 6   probleem           131 non-null    object
 7   oorzaak            131 non-null    object
 8   werken_gepland     138 non-null    object
 9   start_werken       138 non-null    object
 10  bezocht            78 non-null     object
 11  opmerkingen        29 non-null     object
 12  beelden            20 non-null     object
 13  beheerder          1767 non-null   object
 14  beheer             1668 non-null   object
 15  toestandcode       1629 non-null   object
 16  LPK                42 non-null     object


In [561]:
len(frame[frame.hoogproblematisch == 'ja'].index)

607

In [562]:
len(frame.index)

1767

In [563]:
len(frame[frame.hoogproblematisch == 'ja'].index) / len(frame.index) * 100

34.352009054895305

Zo'n 34 procent van alle bruggen die gerenoveerd moeten worden zijn hoogproblematisch. Laten we dat nog iets meer uitsplitsen naar gewest, en ook naar provincie en beheerder. Daarna kunnen we ook kijken of er gemeenten zijn met een bijzonder hoog aantal hoogproblematische bruggen tov de totale bruggen in onze dataset.

### Brussels Gewest (gesitueerd in, niet in bezit van)

In [2]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.gewest == 'Brussels Gewest')])

NameError: name 'frame' is not defined

In [565]:
len(frame[frame.gewest == 'Brussels Gewest'])

55

In [566]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.gewest == 'Brussels Gewest')]) / len(frame[frame.gewest == 'Brussels Gewest']) * 100

20.0

### Vlaams Gewest (gesitueerd in, niet in beheer van)

In [567]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.gewest == 'Vlaanderen')])

596

In [568]:
len(frame[frame.gewest == 'Vlaanderen'])

1712

In [569]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.gewest == 'Vlaanderen')]) / len(frame[frame.gewest == 'Vlaanderen']) * 100

34.81308411214953

### Bonus: hoeveel bruggen in bezit van het gewest moeten eigenlijk in totaal gerenoveerd worden?
We weten dat Brussel 88 bruggen in beheer heeft. Het Vlaams gewest heeft er 2649. In Brussel hebben we alle bruggen die gerenoveerd moeten worden in de lijst staan, van het Vlaams Gewest alle bruggen, maar kijken we alleen naar toestand 2/3/4.

In [570]:
len(frame[frame.beheerder == 'Brussels Gewest']) / 88 * 100

59.09090909090909

In [571]:
len(frame[((frame.beheerder == 'AWV') | (frame.beheerder == 'AWV/VWW') & (frame.beheerder == 'Vlaamse Waterweg') | (frame.beheerder == 'DMOW') | (frame.beheerder == 'MDK') | (frame.beheerder == 'AMW/DMOW')) & ((frame.toestandcode == 'Toestand 2') | (frame.toestandcode == 'Toestand 3') | (frame.toestandcode == 'Toestand 4'))]) / 2649 * 100

42.46885617214043

In het Brussels gewest moeten dus procentueel gezien meer bruggen gerenoveerd worden dan het procentueel aantal bruggen dat het Vlaams Gewest beheert, maar de bruggen in het Brussels gewest zijn er wel iets beter aan toe.

### Provincies

In [572]:
frame.provincie.unique()

array(['Antwerpen', 'Vlaams Brabant', 'Oost-Vlaanderen',
       'West-Vlaanderen', 'Limburg', 'Brussel'], dtype=object)

In [573]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.provincie == 'Antwerpen')]) / len(frame[frame.provincie == 'Antwerpen']) * 100

31.48148148148148

In [574]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.provincie == 'Vlaams Brabant')]) / len(frame[frame.provincie == 'Vlaams Brabant']) * 100

26.545454545454543

In [575]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.provincie == 'Oost-Vlaanderen')]) / len(frame[frame.provincie == 'Oost-Vlaanderen']) * 100

46.913580246913575

In [576]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.provincie == 'West-Vlaanderen')]) / len(frame[frame.provincie == 'West-Vlaanderen']) * 100

33.85300668151447

In [577]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.provincie == 'Limburg')]) / len(frame[frame.provincie == 'Limburg']) * 100

30.24390243902439

In [578]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.provincie == 'Brussel')]) / len(frame[frame.provincie == 'Brussel']) * 100

20.0

Wat valt op? Oost-Vlaanderen is er procentueel gezien behoorlijk slecht aan toe. Waar de andere provincies rond de 20-34 procent van de te renoveren bruggen in problematische staat zijn, is dat in Oost-Vlaanderen bijna de helft van alle te renoveren bruggen. Dat is opvallend, zeker als we kijken naar welke provincie de meeste te renoveren bruggen heeft:

In [579]:
frame.groupby('provincie').idnr.count().sort_values(ascending=False)

provincie
West-Vlaanderen    449
Oost-Vlaanderen    405
Antwerpen          378
Vlaams Brabant     275
Limburg            205
Brussel             55
Name: idnr, dtype: int64

West-Vlaanderen heeft als provincie de meeste te renoveren bruggen in ons databestand, dan Oost-Vlaanderen, dan Antwerpen. Brussel heeft weinig bruggen, minder dan 100.

In [580]:
frame[frame.hoogproblematisch == 'ja'].provincie.value_counts()

Oost-Vlaanderen    190
West-Vlaanderen    152
Antwerpen          119
Vlaams Brabant      73
Limburg             62
Brussel             11
Name: provincie, dtype: int64

Ook in absolute zin heeft Oost-Vlaanderen dus de meeste hoogproblematische bruggen. 

#### Beheerder

In [382]:
frame.beheerder.unique()

array(['AWV', 'AWV/VWW', 'Vlaamse Waterweg', 'DMOW', 'MDK', 'AWV/DMOW',
       'Gemeente', 'Brussels Gewest', 'Infrabel'], dtype=object)

In [581]:
len(frame[(frame.hoogproblematisch == 'ja') & ((frame.beheerder == 'AWV') | (frame.beheerder == 'AWV/VWW') & (frame.beheerder == 'Vlaamse Waterweg') | (frame.beheerder == 'DMOW') | (frame.beheerder == 'MDK') | (frame.beheerder == 'AMW/DMOW'))]) / len(frame[(frame.beheerder == 'AWV') | (frame.beheerder == 'AWV/VWW') & (frame.beheerder == 'Vlaamse Waterweg') | (frame.beheerder == 'DMOW') | (frame.beheerder == 'MDK') | (frame.beheerder == 'AMW/DMOW')]) * 100

35.41666666666667

In [582]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.beheerder == 'Gemeente')]) / len(frame[frame.beheerder == 'Gemeente']) * 100

8.108108108108109

In [583]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.beheerder == 'Brussels Gewest')]) / len(frame[frame.beheerder == 'Brussels Gewest']) * 100

21.153846153846153

In [389]:
len(frame[(frame.hoogproblematisch == 'ja') & (frame.beheerder == 'Infrabel')]) / len(frame[frame.beheerder == 'Infrabel']) * 100

0.0

Noot: van Infrabel weten we er 19 hoogproblematische bruggen zijn, maar we weten nog niet hoeveel te renoveren bruggen Infrabel heeft. Die informatie is niet compleet in ons databestand. Daarom moet die 0 procent genegeerd worden. De gemeentelijke gegevens zijn niet helemaal compleet: een tiental gemeenten heeft nog niet op ons verzoek om informatie gereageerd. Het percentage zou dus nog wat hoger kunnen worden. Gemeenten hebben geen algemene standaarden voor of een brug hoogproblematisch is of niet, dus we zijn afgegaan op kenmerken die de gemeenten ons gaven over de problematische staat van de brug. Ook nu komt het Vlaams Gewest duidelijk als meest problematisch uit de bus.

### Bonus: waar staan al die bruggen die deze beheerders moeten renoveren?

In [584]:
frame[((frame.beheerder == 'AWV') | (frame.beheerder == 'AWV/VWW') & (frame.beheerder == 'Vlaamse Waterweg') | (frame.beheerder == 'DMOW') | (frame.beheerder == 'MDK') | (frame.beheerder == 'AMW/DMOW'))].provincie.value_counts()

West-Vlaanderen    291
Antwerpen          266
Vlaams Brabant     236
Oost-Vlaanderen    227
Limburg            132
Name: provincie, dtype: int64

In [585]:
frame[(frame.beheerder == 'Gemeente')].provincie.value_counts()

Oost-Vlaanderen    13
West-Vlaanderen     9
Vlaams Brabant      8
Limburg             3
Antwerpen           3
Brussel             1
Name: provincie, dtype: int64

In [586]:
frame[(frame.beheerder == 'Infrabel')].provincie.value_counts()

Antwerpen    8
Brussel      2
Name: provincie, dtype: int64

Let op: het gaat hier om alle te renoveren bruggen, niet alleen hoogproblematische bruggen. Het Vlaams Gewest renoveert vooral bruggen in West-Vlaanderen en Antwerpen, de gemeentes vooral in Oost-Vlaanderen, Infrabel vooral in Antwerpen (die data is nog niet volledig)

### Gemeenten

In [587]:
frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1767 entries, 0 to 2901
Data columns (total 22 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   idnr               1767 non-null   object
 1   gemeente           1767 non-null   object
 2   (deelgemeente)     771 non-null    object
 3   provincie          1767 non-null   object
 4   gewest             1767 non-null   object
 5   brugnaam           1767 non-null   object
 6   probleem           131 non-null    object
 7   oorzaak            131 non-null    object
 8   werken_gepland     138 non-null    object
 9   start_werken       138 non-null    object
 10  bezocht            78 non-null     object
 11  opmerkingen        29 non-null     object
 12  beelden            20 non-null     object
 13  beheerder          1767 non-null   object
 14  beheer             1668 non-null   object
 15  toestandcode       1629 non-null   object
 16  LPK                42 non-null     object


In [588]:
frame['totaal_gemeente'] = frame.groupby('gemeente')['brugnaam'].transform('count')

In [589]:
frame.head()

idnr   gemeente (deelgemeente)       provincie      gewest  \
0  00130101  Wommelgem      Wommelgem       Antwerpen  Vlaanderen   
1  00130041     Deurne         Deurne       Antwerpen  Vlaanderen   
2  00130031     Deurne         Deurne       Antwerpen  Vlaanderen   
3  22600241   Arendonk       Arendonk       Antwerpen  Vlaanderen   
4  10000541   Zaventem       Zaventem  Vlaams Brabant  Vlaanderen   

                     brugnaam  \
0  Brug B5 - Wijnegemsteenweg   
1      Brug 2 - Sterckshoflei   
2    Brug 1 - Cornelissenlaan   
3           Brug 5 - Arendonk   
4  Brug B16 Hector Hennaulaan   

                                            probleem      oorzaak  \
0  Vallende brokstukken. Afgesloten voor alle gem...     brokkelt   
1               Jaarlijkse opvolging en waterpassing    technisch   
2  3-maandelijke inspectie, 6-maandelijkse waterp...    technisch   
3                         Renovatieplannen in opmaak     onbekend   
4             Moet vernieuwd worden bij verbreden R0  groter plan   

  werken_gepland start_werken  ... beheerder     beheer toestandcode  \
0            Nee     onbekend  ...       AWV   volledig   Toestand 4   
1             Ja         2021  ...       AWV   volledig   Toestand 4   
2             Ja         2022  ...       AWV   volledig   Toestand 4   
3             Ja         2020  ...   AWV/VWW  bovenbouw   Toestand 4   
4             Ja         2022  ...       AWV   volledig   Toestand 4   

                             LPK prev_year last_year difference  \
0  1e prioriteit - te vernieuwen      2019      2020          1   
1  2e prioriteit - te vernieuwen      2014      2017          3   
2  1e prioriteit - te vernieuwen      2016      2019          3   
3  1e prioriteit - te herstellen      2014      2017          3   
4  1e prioriteit - te vernieuwen      2015      2018          3   

  hoogproblematisch lijst_sinds totaal_gemeente  
0                ja        2007               5  
1                ja        2003               2  
2                ja        2003               2  
3                ja        2011               2  
4                ja        2016              15  

[5 rows x 23 columns]

In [590]:
probleem = frame.groupby(['gemeente', 'hoogproblematisch']).size().reset_index()

In [591]:
probleem = probleem.rename(columns={0: 'totaal_probleem'})

In [592]:
probleem = probleem[probleem.hoogproblematisch == 'ja']

In [593]:
probleem = probleem[['gemeente', 'totaal_probleem']]

In [594]:
probleem.head()

gemeente  totaal_probleem
0        Aalst               12
3       Aalter                2
6     Aarschot                2
9     Affligem                1
12  Alveringem                2

In [595]:
gemeentes = frame.merge(probleem, on='gemeente')

In [596]:
gemeentes['percentage'] = gemeentes['totaal_probleem'] / gemeentes['totaal_gemeente'] * 100

In [597]:
gemeentes['percentage'] = gemeentes.percentage.astype(int)

In [598]:
gemeentes.groupby('gemeente')['totaal_gemeente'].max().sort_values(ascending=False).head(10)

gemeente
Gent         118
Antwerpen    112
Brugge        73
Kortrijk      56
Mechelen      47
Hasselt       29
Genk          27
Leuven        25
Machelen      22
Oostkamp      21
Name: totaal_gemeente, dtype: int64

Gent heeft de meeste te renoveren bruggen, namelijk 118. Verder heeft ook Antwerpen een hoog aantal te renoveren bruggen. 

In [599]:
gemeentes.groupby('gemeente')['totaal_probleem'].max().sort_values(ascending=False).head(10)

gemeente
Gent         52
Antwerpen    45
Kortrijk     28
Brugge       21
Aalst        12
Mechelen     12
Genk         11
Halle         9
Harelbeke     8
Hasselt       8
Name: totaal_probleem, dtype: int64

Gent heeft ook de meeste hoogproblematische bruggen, namelijk 52. We zien ook dat Kortrijk meer hoogproblematische bruggen heeft dan Brugge.

In [600]:
gemeentes.groupby('gemeente')['percentage'].max().sort_values(ascending=False).head(10)

gemeente
Zwijndrecht               100
Galmaarden                100
Poperinge                 100
Meise                     100
Lede                      100
Vleteren                  100
Langemark - Poelkapell    100
Kluisbergen               100
Roosdaal                  100
Gingelom                  100
Name: percentage, dtype: int64

De top tien bestaat geheel uit gemeentes waar álle te renoveren bruggen problematisch zijn. Laten we een sanity check doen.

In [411]:
frame[frame.gemeente == 'Zwijndrecht']

idnr     gemeente (deelgemeente)  provincie      gewest  \
145  00141071  Zwijndrecht            NaN  Antwerpen  Vlaanderen   
399  00141081  Zwijndrecht         Burcht  Antwerpen  Vlaanderen   

                                   brugnaam probleem oorzaak werken_gepland  \
145      BRUG-4-KRIJGSBAAN-(MILITAIRE-BAAN)      NaN     NaN            NaN   
399  BRUG-OVER-STATIESTRAAT-PASTOOR-COPLAAN      NaN     NaN            NaN   

    start_werken  ... beheerder    beheer toestandcode  LPK prev_year  \
145          NaN  ...       AWV  volledig   Toestand 3  NaN      2015   
399          NaN  ...       AWV  volledig   Toestand 3  NaN      2017   

    last_year difference hoogproblematisch lijst_sinds totaal_gemeente  
145      2018          3                ja         nan               2  
399      2019          2                ja         nan               2  

[2 rows x 23 columns]

In [412]:
frame[frame.gemeente == 'Lede']

idnr gemeente (deelgemeente)        provincie      gewest  \
1152  00100721     Lede       Oordegem  Oost-Vlaanderen  Vlaanderen   
1153  00100731     Lede       Oordegem  Oost-Vlaanderen  Vlaanderen   

                                  brugnaam probleem oorzaak werken_gepland  \
1152  BRUG-XCIV-OVER-E40-IN-STICHELENDRIES      NaN     NaN            NaN   
1153  BRUG-25-IN-E40-OVER-DE-KLUIZENSTRAAT      NaN     NaN            NaN   

     start_werken  ... beheerder    beheer toestandcode  LPK prev_year  \
1152          NaN  ...       AWV  volledig   Toestand 3  NaN      2015   
1153          NaN  ...       AWV  volledig   Toestand 3  NaN      2014   

     last_year difference hoogproblematisch lijst_sinds totaal_gemeente  
1152      2018          3                ja         nan               2  
1153      2017          3                ja         nan               2  

[2 rows x 23 columns]

In [413]:
frame[frame.gemeente == 'Gingelom']

idnr  gemeente (deelgemeente) provincie      gewest  brugnaam  \
686  00030731  Gingelom     Montenaken   Limburg  Vlaanderen  BRUG-B73   
687  00030741  Gingelom     Montenaken   Limburg  Vlaanderen  BRUG-B74   

    probleem oorzaak werken_gepland start_werken  ... beheerder    beheer  \
686      NaN     NaN            NaN          NaN  ...       AWV  volledig   
687      NaN     NaN            NaN          NaN  ...       AWV  volledig   

    toestandcode  LPK prev_year last_year difference hoogproblematisch  \
686   Toestand 3  NaN      2014      2018          4                ja   
687   Toestand 3  NaN      2014      2017          3                ja   

    lijst_sinds totaal_gemeente  
686         nan               2  
687         nan               2  

[2 rows x 23 columns]

Ook de sanity check bevestigt dat dit allemaal gemeenten zijn waar de bruggen hoogproblematisch zijn, dat wil zeggen: toestand 3 of 4 in Vlaanderen, toestand A of B in Brussel.

In [601]:
gemeentes[gemeentes.percentage == 100].gemeente.nunique()

18

In [602]:
gemeentes[gemeentes.percentage == 100].gemeente.unique()

array(['Deurne', 'Edegem', 'Zwijndrecht', 'Balen', 'Duffel', 'Dessel',
       'Zoersel', 'Gingelom', 'Halen', 'Lede', 'Kluisbergen', 'Ronse',
       'Roosdaal', 'Meise', 'Galmaarden', 'Vleteren',
       'Langemark - Poelkapell', 'Poperinge'], dtype=object)

In totaal zijn er 18 gemeenten waar dit speelt. 

In [603]:
gemeentes[(gemeentes.percentage > 80) & (gemeentes.percentage < 100)].gemeente.unique()

array(['Overijse', 'Erpe-Mere', 'Deinze', 'Nazareth', 'Kruishoutem',
       'Waasmunster'], dtype=object)

In [604]:
gemeentes[(gemeentes.percentage > 50) & (gemeentes.percentage < 100)].gemeente.nunique()

28

In [605]:
gemeentes[(gemeentes.percentage > 50) & (gemeentes.percentage < 100)].gemeente.unique()

array(['Wommelgem', 'Overijse', 'Halle', 'Erpe-Mere', 'Waregem', 'Aalst',
       'Deinze', 'Westerlo', 'Laakdal', 'Zandhoven', 'Dilsen-Stokkem',
       'Voeren', 'Ninove', 'Dendermonde', 'Lokeren', 'Oudenaarde',
       'Nazareth', 'Lochristi', 'Wachtebeke', 'Kruishoutem', 'Zele',
       'Waasmunster', 'Laarne', 'Tervuren', 'Harelbeke', 'Torhout',
       'Izegem', 'Koksijde'], dtype=object)

In [606]:
gemeentes[gemeentes.percentage > 50].gemeente.nunique()

46

Het valt op dat hoewel Gent in de absolute zin de meeste hoogproblematische en de meeste te renoveren bruggen heeft, het percentage lager uitvalt: minder dan 50 procent van de bruggen. Voor 28 gemeenten is meer dan de helft van de te renoveren bruggen hoogproblematisch. Voor 6 gemeenten is dat meer dan 80 procent.

## Wie beheert die hoogproblematische bruggen in die gemeenten?
Eerst bekijken we de lijst van 100 procent, dan van hoger dan 80, dan van hoger dan 50.

In [607]:
gemeentes[gemeentes.percentage == 100].beheerder.value_counts()

AWV                 21
Vlaamse Waterweg     6
AWV/VWW              2
Gemeente             1
Name: beheerder, dtype: int64

In [608]:
gemeentes[(gemeentes.percentage == 100) & (gemeentes.beheerder == 'Gemeente')]

idnr    gemeente (deelgemeente)       provincie      gewest  \
1326  nan  Galmaarden            NaN  Vlaams Brabant  Vlaanderen   

                                  brugnaam              probleem   oorzaak  \
1326  Brug over de Mark bij Stationsstraat  betonrot, raming 65K  betonrot   

     werken_gepland start_werken  ... toestandcode  LPK prev_year last_year  \
1326             Ja     onbekend  ...          NaN  NaN       nan       nan   

     difference hoogproblematisch lijst_sinds totaal_gemeente totaal_probleem  \
1326        nan                ja         nan               2               2   

     percentage  
1326        100  

[1 rows x 25 columns]

Er is maar één geval waar de hoogproblematische brug wordt beheerd door de gemeente zelf. In alle andere gevallen gaat het om bruggen in het beheer van het Vlaams gewest.

In [609]:
gemeentes[(gemeentes.percentage > 80) & (gemeentes.percentage < 100)].beheerder.value_counts()

AWV                 32
AWV/VWW              4
Vlaamse Waterweg     3
Name: beheerder, dtype: int64

In [610]:
gemeentes[(gemeentes.percentage > 50) & (gemeentes.percentage < 80)].beheerder.value_counts()

AWV                 90
Vlaamse Waterweg    35
AWV/VWW             13
Gemeente             1
Name: beheerder, dtype: int64

In [611]:
gemeentes[(gemeentes.percentage > 50) & (gemeentes.percentage < 80) & (gemeentes.beheerder == 'Gemeente')]

idnr gemeente (deelgemeente) provincie      gewest  \
930  nan   Voeren            NaN   Limburg  Vlaanderen   

                              brugnaam  \
930  Brug over "De Gulp" in Remersdaal   

                                              probleem    oorzaak  \
930  is bouwvallig, afgesloten voor gemotoriseerd v...  verouderd   

    werken_gepland start_werken  ... toestandcode  LPK prev_year last_year  \
930             Ja         2020  ...          NaN  NaN       nan       nan   

    difference hoogproblematisch lijst_sinds totaal_gemeente totaal_probleem  \
930        nan                ja         nan               3               2   

    percentage  
930         66  

[1 rows x 25 columns]

Dit betekent dat van de 46 gemeenten waar meer dan de helft van de bruggen die gerenoveerd moeten worden hoogproblematisch is, de meeste gemeenten daar zelf geen beheerder van zijn en het onderhoud dus voor rekening komt van het Gewest. We weten uit onze eigen bevraging dat gemeenten daardoor ook vaak niet op de hoogte zijn van de staat van die bruggen. Dat brengt ons op de vraag hoe lang die gemeenten dan moeten wachten tot zoiets wordt gerenoveerd.

## Hoe lang staan bruggen op de lijst voor ze worden gerenoveerd?
Deze informatie hebben we alleen voor de bruggen van het Vlaams Gewest die er het ergst aan toe zijn.

In [500]:
frame.lijst_sinds.unique()

array(['2007', '2003', '2011', '2016', '2018', '2009', '2019', '1997',
       '1999', '2001', '2013', '2014', '2017', '201', 'nan'], dtype=object)

In [504]:
frame.lijst_sinds = frame.lijst_sinds.replace('201', '2001')

In [505]:
frame.last_year = frame.last_year.replace('202', '2020')

In [520]:
frame.prev_year = frame.prev_year.replace('201', '2010')

In [506]:
frame.last_year.unique()

array(['2020', '2017', '2019', '2018', '2012', '2016', '2014', '2015',
       'nan', '2011', '2013'], dtype=object)

In [521]:
vlaams = frame[frame.lijst_sinds != 'nan']

In [508]:
vlaams.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 42 entries, 0 to 41
Data columns (total 22 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   idnr               42 non-null     object
 1   gemeente           42 non-null     object
 2   (deelgemeente)     31 non-null     object
 3   provincie          42 non-null     object
 4   gewest             42 non-null     object
 5   brugnaam           42 non-null     object
 6   probleem           42 non-null     object
 7   oorzaak            42 non-null     object
 8   werken_gepland     42 non-null     object
 9   start_werken       42 non-null     object
 10  bezocht            42 non-null     object
 11  opmerkingen        18 non-null     object
 12  beelden            17 non-null     object
 13  beheerder          42 non-null     object
 14  beheer             42 non-null     object
 15  toestandcode       42 non-null     object
 16  LPK                42 non-null     object
 17 

Let op: dit zijn niet alle bruggen die er het slechtst aan toe zijn. Dat zijn er meer:

In [496]:
len(frame[frame.toestandcode == 'Toestand 4'])

46

In [509]:
frame[(frame.toestandcode == 'Toestand 4') & (frame.lijst_sinds == 'nan')]

idnr    gemeente (deelgemeente)        provincie      gewest  \
1464  25310063        Gent            NaN  Oost-Vlaanderen  Vlaanderen   
1799  90290241       Diest            NaN   Vlaams Brabant  Vlaanderen   
2180  25400041   Oudenburg            NaN  West-Vlaanderen  Vlaanderen   
2210  27300031  Lo-Reninge             Lo  West-Vlaanderen  Vlaanderen   

                                          brugnaam probleem oorzaak  \
1464                    VOETBRUG-OVER-TOLHUISSLUIS      NaN     NaN   
1799                         BRUG-OVER-ZWARTE-BEEK      NaN     NaN   
2180  GROTE-PLASSENDALEBRUG-KANAAL-BRUGGE-OOSTENDE      NaN     NaN   
2210                                KELLENAARSBRUG      NaN     NaN   

     werken_gepland start_werken  ... beelden         beheerder    beheer  \
1464            NaN          NaN  ...     NaN  Vlaamse Waterweg  volledig   
1799            NaN          NaN  ...     NaN               AWV  volledig   
2180            NaN          NaN  ...     NaN  Vlaamse Waterweg  volledig   
2210            NaN          NaN  ...     NaN  Vlaamse Waterweg  volledig   

     toestandcode  LPK prev_year last_year difference hoogproblematisch  \
1464   Toestand 4  NaN      2014      2017          3                ja   
1799   Toestand 4  NaN      2013      2018          5                ja   
2180   Toestand 4  NaN      2016      2019          3                ja   
2210   Toestand 4  NaN      2014      2017          3                ja   

     lijst_sinds  
1464         nan  
1799         nan  
2180         nan  
2210         nan  

[4 rows x 22 columns]

Deze bruggen zijn dus sinds december 2019 (de laatste versie van de lijst van AWV) op de lijst met prioritaire kunstwerken gekomen. Dat is frappant, want we zien dat sommige van deze kunstwerken voor het laatst zijn gecontroleerd in 2018 of zelfs 2017. De vraag is waarom ze dan niet al eerder op de lijst stonden.

In [522]:
vlaams.lijst_sinds = vlaams.lijst_sinds.astype(int)

In [511]:
len(vlaams[vlaams.lijst_sinds < 2000])

5

In [512]:
vlaams[vlaams.lijst_sinds < 2000]

idnr  gemeente                (deelgemeente)        provincie  \
12  00100471      Gent  Sint-Denijs-Westrem - Afsnee  Oost-Vlaanderen   
13  00100451      Gent                       Drongen  Oost-Vlaanderen   
14  00100551      Gent                    Zwijnaarde  Oost-Vlaanderen   
15  00100561      Gent                    Zwijnaarde  Oost-Vlaanderen   
16  00100681  Wetteren                       Westrem  Oost-Vlaanderen   

        gewest                                    brugnaam  \
12  Vlaanderen         Goedingebrug / Brug 12 over de Leie   
13  Vlaanderen  Brug LXXIV over de A10 aan op- en afrit 13   
14  Vlaanderen                      Slingerviaduct IJ-West   
15  Vlaanderen                      Slingerviaduct MN-Oost   
16  Vlaanderen                         Brug XC over de E40   

                               probleem      oorzaak werken_gepland  \
12                Moet vernieuwd worden     onbekend            Nee   
13  Wordt vernieuwd met complex Drongen  groter plan             Ja   
14                Moet vernieuwd worden     onbekend             Ja   
15                Moet vernieuwd worden     onbekend             Ja   
16                     Wordt afgebroken     onbekend            Nee   

   start_werken  ...        beelden beheerder    beheer toestandcode  \
12     onbekend  ...  Foto en video       AWV  volledig   Toestand 4   
13         2021  ...            Nee       AWV  volledig   Toestand 4   
14         2023  ...            Nee       AWV  volledig   Toestand 4   
15         2024  ...            Nee       AWV  volledig   Toestand 4   
16     onbekend  ...  Foto en video       AWV  volledig   Toestand 4   

                              LPK prev_year last_year difference  \
12  1e prioriteit - te vernieuwen      2013      2016          3   
13  2e prioriteit - te vernieuwen      2017      2020          3   
14  2e prioriteit - te herstellen      2014      2017          3   
15  2e prioriteit - te herstellen      2014      2017          3   
16  1e prioriteit - te vernieuwen       201      2018          8   

   hoogproblematisch lijst_sinds  
12                ja        1997  
13                ja        1997  
14                ja        1997  
15                ja        1997  
16                ja        1999  

[5 rows x 22 columns]

In [513]:
len(vlaams[vlaams.lijst_sinds < 2005])

11

In [514]:
vlaams[vlaams.lijst_sinds < 2005]

idnr   gemeente                (deelgemeente)        provincie  \
1   00130041     Deurne                        Deurne        Antwerpen   
2   00130031     Deurne                        Deurne        Antwerpen   
12  00100471       Gent  Sint-Denijs-Westrem - Afsnee  Oost-Vlaanderen   
13  00100451       Gent                       Drongen  Oost-Vlaanderen   
14  00100551       Gent                    Zwijnaarde  Oost-Vlaanderen   
15  00100561       Gent                    Zwijnaarde  Oost-Vlaanderen   
16  00100681   Wetteren                       Westrem  Oost-Vlaanderen   
17  00100461       Gent                       Drongen  Oost-Vlaanderen   
18  00100441       Gent                       Drongen  Oost-Vlaanderen   
19  00100821  Erpe-Mere                          Erpe  Oost-Vlaanderen   
38  23000221      Halle                         Halle   Vlaams Brabant   

        gewest                                    brugnaam  \
1   Vlaanderen                      Brug 2 - Sterckshoflei   
2   Vlaanderen                    Brug 1 - Cornelissenlaan   
12  Vlaanderen         Goedingebrug / Brug 12 over de Leie   
13  Vlaanderen  Brug LXXIV over de A10 aan op- en afrit 13   
14  Vlaanderen                      Slingerviaduct IJ-West   
15  Vlaanderen                      Slingerviaduct MN-Oost   
16  Vlaanderen                         Brug XC over de E40   
17  Vlaanderen                   Brug LXXV - Solvijnsdreef   
18  Vlaanderen                      Brug LXXIII in Drongen   
19  Vlaanderen     Brug 30 in de A10 over de spoorlijn 50A   
38  Vlaanderen                                Bospoortbrug   

                                             probleem      oorzaak  \
1                Jaarlijkse opvolging en waterpassing    technisch   
2   3-maandelijke inspectie, 6-maandelijkse waterp...    technisch   
12                              Moet vernieuwd worden     onbekend   
13                Wordt vernieuwd met complex Drongen  groter plan   
14                              Moet vernieuwd worden     onbekend   
15                              Moet vernieuwd worden     onbekend   
16                                   Wordt afgebroken     onbekend   
17                Wordt vernieuwd met complex Drongen  groter plan   
18                Wordt vernieuwd met complex Drongen  groter plan   
19                              Moet vernieuwd worden     onbekend   
38                        Wordt op termijn afgebroken     onbekend   

   werken_gepland start_werken  ...        beelden         beheerder  \
1              Ja         2021  ...            NaN               AWV   
2              Ja         2022  ...            NaN               AWV   
12            Nee     onbekend  ...  Foto en video               AWV   
13             Ja         2021  ...            Nee               AWV   
14             Ja         2023  ...            Nee               AWV   
15             Ja         2024  ...            Nee               AWV   
16            Nee     onbekend  ...  Foto en video               AWV   
17             Ja         2021  ...  Foto en video               AWV   
18             Ja         2021  ...  Foto en video               AWV   
19             Ja         2022  ...            Nee               AWV   
38            Nee     onbekend  ...            NaN  Vlaamse Waterweg   

      beheer toestandcode                            LPK prev_year last_year  \
1   volledig   Toestand 4  2e prioriteit - te vernieuwen      2014      2017   
2   volledig   Toestand 4  1e prioriteit - te vernieuwen      2016      2019   
12  volledig   Toestand 4  1e prioriteit - te vernieuwen      2013      2016   
13  volledig   Toestand 4  2e prioriteit - te vernieuwen      2017      2020   
14  volledig   Toestand 4  2e prioriteit - te herstellen      2014      2017   
15  volledig   Toestand 4  2e prioriteit - te herstellen      2014      2017   
16  volledig   Toestand 4  1e prioriteit - te vernieuwen       201      2018   
17  volledig   Toestand 4  2

In [515]:
len(vlaams[vlaams.lijst_sinds < 2015])

21

In [516]:
vlaams[vlaams.lijst_sinds < 2000].gemeente.value_counts()

Gent        4
Wetteren    1
Name: gemeente, dtype: int64

In [517]:
vlaams[vlaams.lijst_sinds < 2005].gemeente.value_counts()

Gent         6
Deurne       2
Wetteren     1
Halle        1
Erpe-Mere    1
Name: gemeente, dtype: int64

In [518]:
vlaams[vlaams.lijst_sinds < 2015].gemeente.value_counts()

Gent         9
Kortrijk     2
Zaventem     2
Deurne       2
Wommelgem    1
Waregem      1
Erpe-Mere    1
Wetteren     1
Halle        1
Arendonk     1
Name: gemeente, dtype: int64

Er zijn 5 bruggen die al langer dan 20 jaar op de lijst staan, 21 bruggen die al langer dan 5 jaar op de lijst staan. Die lijst wordt telkens aangevoerd voor Gent. Van de vijf bruggen die al langer dan 20 jaar op de lijst staan, komen er 4 uit Gent, één uit Wetteren. Opvallend: de brug uit Wetteren is in 2018 voor het laatst gecontroleerd, de keer daarvoor was in 2010. Terwijl de brug er dus al meer dan 20 jaar op staat. Dat is een ander interessant gegeven: hoe vaak worden deze bruggen, die toch in kritieke toestand zijn, gecontroleerd? 

In [523]:
vlaams.difference.value_counts()

3    29
6     4
4     4
8     2
5     2
1     1
Name: difference, dtype: int64

In [525]:
vlaams[vlaams.difference == '6']

idnr    gemeente     (deelgemeente)        provincie      gewest  \
7   00040201    Overijse           Overijse   Vlaams Brabant  Vlaanderen   
10  00120046  Grimbergen  Strombeek - Bever   Vlaams Brabant  Vlaanderen   
11  00120047  Grimbergen  Strombeek - Bever   Vlaams Brabant  Vlaanderen   
28  25200462       Aalst                NaN  Oost-Vlaanderen  Vlaanderen   

                       brugnaam                     probleem      oorzaak  \
7   Brug O9 over de Bredestraat            Werken zijn bezig     onbekend   
10                     BRUG-W20                  Doorbuiging  doorbuiging   
11      BRUG-W21-IN-A12-OVER-R0                  Doorbuiging  doorbuiging   
28                SINT-ANNABRUG  Uitvoering voorzien in 2020     onbekend   

   werken_gepland start_werken  ... beelden         beheerder    beheer  \
7              Ja         2020  ...     NaN               AWV  volledig   
10            Nee     onbekend  ...     NaN               AWV  volledig   
11            Nee     onbekend  ...     NaN               AWV  volledig   
28             Ja         2020  ...     NaN  Vlaamse Waterweg  volledig   

   toestandcode                            LPK prev_year last_year difference  \
7    Toestand 4  1e prioriteit - te vernieuwen      2012      2018          6   
10   Toestand 4  1e prioriteit - te vernieuwen      2012      2018          6   
11   Toestand 4  1e prioriteit - te vernieuwen      2012      2018          6   
28   Toestand 4  1e prioriteit - te herstellen      2011      2017          6   

   hoogproblematisch lijst_sinds  
7                 ja        2016  
10                ja        2019  
11                ja        2019  
28                ja        2018  

[4 rows x 22 columns]

In [526]:
vlaams[vlaams.difference == '8']

idnr  gemeente (deelgemeente)        provincie      gewest  \
16  00100681  Wetteren        Westrem  Oost-Vlaanderen  Vlaanderen   
20  00140681      Gent     Gentbrugge  Oost-Vlaanderen  Vlaanderen   

                                 brugnaam              probleem   oorzaak  \
16                    Brug XC over de E40      Wordt afgebroken  onbekend   
20  Viaduct van Gentbrugge 80/O in de A14  Vallende brokstukken  brokkelt   

   werken_gepland start_werken  ...        beelden beheerder    beheer  \
16            Nee     onbekend  ...  Foto en video       AWV  volledig   
20             Ja         2020  ...  Foto en video       AWV  volledig   

   toestandcode                            LPK prev_year last_year difference  \
16   Toestand 4  1e prioriteit - te vernieuwen      2010      2018          8   
20   Toestand 4  2e prioriteit - te herstellen      2011      2019          8   

   hoogproblematisch lijst_sinds  
16                ja        1999  
20                ja        2009  

[2 rows x 22 columns]

Naast de brug in Wetteren is dus ook de brug in Gentbrugge, die nu onder handen wordt genomen, er één waar 8 jaar over gedaan is om die te controleren, hoewel die al sinds 2009 op de lijst staat. Ook opvallend: er staan een aantal bruggen pas vrij recent op de lijst (2018, 2019) maar daar duurde het zes jaar tussen de voorlaatste en de laatste controle. Ander opvallend punt: bij twee van deze bruggen zijn nog altijd geen werken gepland, namelijk de bruggen in Grimbergen. Het heeft dus lang geduurd voor ze gecontroleerd werden, daarna zijn ze op de lijst gekomen, maar er is nog altijd geen datum gepland om ze te renoveren. Zeker opvallend omdat ze allemaal eerste prioriteit te vernieuwen zijn.

## Hoe vaak worden de bruggen in het Vlaams gewest überhaupt gecontroleerd?
Hiervoor wil ik naar de hele datset voor het Vlaams gewest, dus ook voor de bruggen die nog in goede staat zouden verkeren. Het is namelijk interessant om na te gaan of er een dusdanige achterstand in de controles is, waardoor ook bruggen die nu een toestand '1' hebben, misschien al in slechtere staat verkeren.

In [10]:
df.last_year = df.last_year.replace('202', '2020')

In [11]:
df.difference = df.difference.replace('202', '2020')

In [12]:
df.prev_year = df.prev_year.replace('201', '2010')

In [532]:
df.difference.value_counts()

3       1633
4        483
5        271
nan      141
6         54
2         52
2018      25
2019      23
2017      14
2015      12
7         10
2016       8
1          8
9          4
8          3
2020       3
15         1
2011       1
2014       1
2012       1
Name: difference, dtype: int64

De meeste bruggen worden om de drie jaar gecontroleerd, maar niet allemaal. NB: bruggen met een difference > 2000 zijn nooit eerder gecontroleerd geweest. Het is interessant om te zien in welke staat die bruggen verkeren. Er zijn ook veel bruggen die om 4 of 5 jaar gecontroleerd worden. Opvallend: bruggen die maar met tussenpozen van elke 7 jaar of langer gecontroleerd worden.

In [534]:
df[df.difference == '15']

idnr      gemeente        (deelgemeente) provincie      gewest  \
921  22300221  Maasmechelen  Mechelen-Aan-De-Maas   Limburg  Vlaanderen   

                                    brugnaam probleem oorzaak werken_gepland  \
921  BRUG-MAASMECHELEN-AAN-DOKTER-HAUBENLAAN      NaN     NaN            NaN   

    start_werken  ... beelden beheerder     beheer toestandcode  LPK  \
921          NaN  ...     NaN   AWV/VWW  bovenbouw   Toestand 3  NaN   

    prev_year last_year difference hoogproblematisch lijst_sinds  
921      2002      2017         15                ja         nan  

[1 rows x 22 columns]

In [541]:
df[df.difference > '6'].toestandcode.value_counts()

Toestand t    17
Toestand 3     5
Toestand 1     4
Toestand 2     3
Toestand 4     2
Name: toestandcode, dtype: int64

In [543]:
df[(df.difference > '6') & (df.toestandcode == 'Toestand 3')]

idnr  gemeente      (deelgemeente)        provincie      gewest  \
1506  00141011     Temse                 NaN  Oost-Vlaanderen  Vlaanderen   
1806  00040141  Overijse            Overijse   Vlaams Brabant  Vlaanderen   
1820  00101091   Dilbeek  Groot - Bijgaarden   Vlaams Brabant  Vlaanderen   
1879  90030091    Tienen                 NaN   Vlaams Brabant  Vlaanderen   
2375  25400162    Brugge                 NaN  West-Vlaanderen  Vlaanderen   

                    brugnaam probleem oorzaak werken_gepland start_werken  \
1506  BRUG-4-LANDMOLENSTRAAT      NaN     NaN            NaN          NaN   
1806   Brug O2 Esdoornenlaan      NaN     NaN            NaN          NaN   
1820               BRUG-CXXI      NaN     NaN            NaN          NaN   
1879        GETE-OVERWELVING      NaN     NaN            NaN          NaN   
2375      KATELIJNEPOORTBRUG      NaN     NaN            NaN          NaN   

      ... beelden beheerder    beheer toestandcode  LPK prev_year last_year  \
1506  ...     NaN       AWV  volledig   Toestand 3  NaN      2011      2018   
1806  ...     NaN       AWV  volledig   Toestand 3  NaN      2011      2018   
1820  ...     NaN       AWV  volledig   Toestand 3  NaN      2011      2018   
1879  ...     NaN       AWV  volledig   Toestand 3  NaN      2011      2018   
2375  ...     NaN       AWV  volledig   Toestand 3  NaN      2017      2017   

     difference hoogproblematisch lijst_sinds  
1506          7                ja         nan  
1806          7                ja         nan  
1820          7                ja         nan  
1879          7                ja         nan  
2375        nan                ja         nan  

[5 rows x 22 columns]

In [549]:
df[(df.difference > '6') & (df.toestandcode == 'Toestand t')]

idnr     gemeente (deelgemeente)        provincie      gewest  \
668   00130233  Grobbendonk    Grobbendonk        Antwerpen  Vlaanderen   
682   10110142     Wijnegem       Wijnegem        Antwerpen  Vlaanderen   
1691  10021504      Beveren          Kallo  Oost-Vlaanderen  Vlaanderen   
1696  21000052     Wetteren       Wetteren  Oost-Vlaanderen  Vlaanderen   
1699  00100592         Gent     Zwijnaarde  Oost-Vlaanderen  Vlaanderen   
2095  21300159       Leuven        Wilsele   Vlaams Brabant  Vlaanderen   
2096  21300163       Leuven        Wilsele   Vlaams Brabant  Vlaanderen   
2098  90260055       Herent            NaN   Vlaams Brabant  Vlaanderen   
2112  92070021   Liedekerke     Liedekerke   Vlaams Brabant  Vlaanderen   
2113  10000438     Machelen       Machelen   Vlaams Brabant  Vlaanderen   
2117  00012011   Londerzeel     Londerzeel   Vlaams Brabant  Vlaanderen   
2118  00120123   Londerzeel     Londerzeel   Vlaams Brabant  Vlaanderen   
2125  00020732       Leuven       Heverlee   Vlaams Brabant  Vlaanderen   
2726  90360047    Harelbeke            NaN  West-Vlaanderen  Vlaanderen   
2727  90360048    Harelbeke            NaN  West-Vlaanderen  Vlaanderen   
2728  90360053    Harelbeke            NaN  West-Vlaanderen  Vlaanderen   
2729  90360054    Harelbeke            NaN  West-Vlaanderen  Vlaanderen   

                                               brugnaam probleem oorzaak  \
668   Brug-O22-brughelft-richt. Antwerp.-over spoorw...      NaN     NaN   
682   Fietsbrug in R11 over Groot Schijn naast duike...      NaN     NaN   
1691             Brug Hollands Complex thv rotonde WLHN      NaN     NaN   
1696                 Fiets- en voetbrug over de Schelde      NaN     NaN   
1699                    Brug in E40 over Tijarm Schelde      NaN     NaN   
2095                                    DIJLEBRUG-NOORD      NaN     NaN   
2096                                     DIJLEBRUG-ZUID      NaN     NaN   
2098                        TUNNEL-N26-WILSELSESTEENWEG      NaN     NaN   
2112                                Brug Affligemstraat      NaN     NaN   
2113                         Brug ontsluiting Uplace-K7      NaN     NaN   
2117                        Kokerbrug O7 Mechelsestraat      NaN     NaN   
2118                          Brug O8 over spoorlijn 53      NaN     NaN   
2125  Fietsbrug over E314 naast de brug WOL-B0-in de...      NaN     NaN   
2726         FIETSTUNNEL-ONDER-AFRIT-N36-KANT-ROESELARE      NaN     NaN   
2727         FIETSTUNNEL-ONDER-OPRIT-N36-KANT-ROESELARE      NaN     NaN   
2728          FIETSTUNNEL-ONDER-OPRIT-N36-KANT-DEERLIJK      NaN     NaN   
2729          FIETSTUNNEL-ONDER-AFRIT-N36-KANT-DEERLIJK      NaN     NaN   

     werken_gepland start_werken  ... beelden         beheerder    beheer  \
668             NaN          NaN  ...     NaN               AWV  volledig   
682             NaN          NaN  ...     NaN               AWV  volledig   
1691            NaN          NaN  ...     NaN               AWV  volledig   
1696            NaN          NaN  ...     NaN  Vlaamse Waterweg  volledig   
1699            NaN          NaN  ...     NaN               AWV  volledig   
2095            NaN          NaN  ...     NaN               AWV  volledig   
2096            NaN          NaN  ...     NaN               AWV  volledig   
2098            NaN          NaN  ...     NaN               AWV  volledig   
2112            NaN          NaN  ...     NaN               AWV  volledig   
2113            NaN          NaN  ...     NaN               AWV  volledig   
2117            NaN          NaN  ...     NaN               AWV  volledig   
2118            NaN          NaN  ...     NaN               AWV  volledig   
2125            NaN          NaN  ...     NaN               AWV  volledig   
2726            NaN          NaN  ...     NaN               AWV  volledig   
2727            NaN          NaN  ...     NaN               AWV  volledig   
2728            NaN          NaN  ...     NaN             

Er zijn vier fietsersbruggen (tunnels) in Harelbeke die tussen 2010 en 2019 helemaal niet gecontroleerd zijn. Daarvan moet de toestand nog bepaald worden. Er zijn ook een aantal bruggen die momenteel toestand 3 hebben en tussen 2011 en 2018 niet zijn gecontroleerd. 

In [550]:
df.toestandcode.value_counts()

Toestand 2    1036
Toestand 1     924
Toestand 3     547
Toestand t      51
Toestand 4      46
Geen beoor       6
Name: toestandcode, dtype: int64

De meeste bruggen in het Vlaams gewest hebben toestandcode 2: lichte gebreken. Renovatie op lange termijn is gewenst in dat geval. Er zijn ook een zestigtal bruggen waarvan geen beoordeling bekend is, of waarvan de toestand nog bepaald moet worden. Waarom niet? Dat is een interessante vraag die nu onbeantwoord blijft.

In [556]:
df[df.toestandcode == 'Geen beoor']

idnr   gemeente (deelgemeente)        provincie      gewest  \
392   01002215  Antwerpen            NaN        Antwerpen  Vlaanderen   
531   00210105     Beerse            NaN        Antwerpen  Vlaanderen   
532   00210107     Beerse            NaN        Antwerpen  Vlaanderen   
2013  90290211     Tienen            NaN   Vlaams Brabant  Vlaanderen   
2073  90190203   Holsbeek            NaN   Vlaams Brabant  Vlaanderen   
2712  08320021   De Panne            NaN  West-Vlaanderen  Vlaanderen   

                                        brugnaam probleem oorzaak  \
392   SPOORWEGKOKER-K1-WEST-IN-R2-OVER-SPOORLIJN      NaN     NaN   
531        DUIKER-23/1-IN-VORSELAAR/LOPENDE-BEEK      NaN     NaN   
532          DUIKER-23/2-IN-VORSELAAR/PULDERBEEK      NaN     NaN   
2013                      BORGGRACHT-OVERWELVING      NaN     NaN   
2073                              BRUG-OVER-VUNT      NaN     NaN   
2712                     BRUG-WEST-DIJKDOORBRAAK      NaN     NaN   

     werken_gepland start_werken  ... beelden beheerder    beheer  \
392             NaN          NaN  ...     NaN       AWV  volledig   
531             NaN          NaN  ...     NaN       AWV  volledig   
532             NaN          NaN  ...     NaN       AWV  volledig   
2013            NaN          NaN  ...     NaN       AWV  volledig   
2073            NaN          NaN  ...     NaN       AWV  volledig   
2712            NaN          NaN  ...     NaN       MDK  volledig   

     toestandcode  LPK prev_year last_year difference hoogproblematisch  \
392    Geen beoor  NaN      2016      2019          3          onbekend   
531    Geen beoor  NaN      2015      2018          3          onbekend   
532    Geen beoor  NaN      2015      2018          3          onbekend   
2013   Geen beoor  NaN      2014      2018          4          onbekend   
2073   Geen beoor  NaN      2012      2018          6          onbekend   
2712   Geen beoor  NaN      2013      2016          3          onbekend   

     lijst_sinds  
392          nan  
531          nan  
532          nan  
2013         nan  
2073         nan  
2712         nan  

[6 rows x 22 columns]

In [48]:
df.beheerder.unique()

array(['AWV', 'AWV/VWW', 'Vlaamse Waterweg', 'DMOW', 'MDK', 'AWV/DMOW',
       'Gemeente', 'Brussels Gewest', 'Infrabel'], dtype=object)

In [49]:
df[(df.provincie == 'Antwerpen') & (df.beheerder != 'Brussels Gewest') & (df.beheerder != 'Gemeente') & (df.beheerder != 'Infrabel')].difference.value_counts()

3       429
4        93
5        30
2        12
2015      9
2017      7
2019      6
2016      6
nan       5
2018      4
8         1
6         1
1         1
Name: difference, dtype: int64

In [50]:
df[(df.provincie == 'Vlaams Brabant') & (df.beheerder != 'Brussels Gewest') & (df.beheerder != 'Gemeente') & (df.beheerder != 'Infrabel')].difference.value_counts()

5       127
3       114
4       111
6        36
nan      13
7         8
2018      5
2019      2
2         1
2017      1
1         1
Name: difference, dtype: int64

In [51]:
df[(df.provincie == 'Oost-Vlaanderen') & (df.beheerder != 'Brussels Gewest') & (df.beheerder != 'Gemeente') & (df.beheerder != 'Infrabel')].difference.value_counts()

3       384
4        63
5        51
2        19
2019      8
2018      8
nan       7
6         7
2020      3
2017      3
8         2
2012      1
7         1
2015      1
2016      1
2014      1
2011      1
1         1
Name: difference, dtype: int64

In [52]:
df[(df.provincie == 'West-Vlaanderen') & (df.beheerder != 'Brussels Gewest') & (df.beheerder != 'Gemeente') & (df.beheerder != 'Infrabel')].difference.value_counts()

3       414
4       150
5        56
2019      5
9         4
6         4
1         4
2         3
2018      3
nan       2
2015      2
2016      1
2017      1
Name: difference, dtype: int64

In [53]:
df[(df.provincie == 'Limburg') & (df.beheerder != 'Brussels Gewest') & (df.beheerder != 'Gemeente') & (df.beheerder != 'Infrabel')].difference.value_counts()

3       292
4        66
2        17
nan      15
5         7
6         6
2018      5
2019      2
2017      2
15        1
7         1
1         1
Name: difference, dtype: int64

In [58]:
df[(df.beheerder != 'Brussels Gewest') & (df.beheerder != 'Gemeente') & (df.beheerder != 'Infrabel') & (df.difference == 'nan') & (df.last_year != 'nan')]

idnr gemeente (deelgemeente)        provincie      gewest  \
1547  25400261   Aalter         Bellem  Oost-Vlaanderen  Vlaanderen   
2375  25400162   Brugge            NaN  West-Vlaanderen  Vlaanderen   

                brugnaam probleem oorzaak werken_gepland start_werken  ...  \
1547          BELLEMBRUG      NaN     NaN            NaN          NaN  ...   
2375  KATELIJNEPOORTBRUG      NaN     NaN            NaN          NaN  ...   

     beelden         beheerder    beheer toestandcode  LPK prev_year  \
1547     NaN  Vlaamse Waterweg  volledig   Toestand 1  NaN      2019   
2375     NaN               AWV  volledig   Toestand 3  NaN      2017   

     last_year difference hoogproblematisch lijst_sinds  
1547      2019        nan               nee         nan  
2375      2017        nan                ja         nan  

[2 rows x 22 columns]

In [33]:
df[((df.difference == '2011') | (df.difference == '2012') | (df.difference == '2014') | (df.difference == '2015') | (df.difference == '2016') | (df.difference == '2017') | (df.difference == '2018') | (df.difference == '2019') | (df.difference == '2020')) & (df.difference != 'nan')].hoogproblematisch.value_counts()

nee         53
onbekend    33
ja           2
Name: hoogproblematisch, dtype: int64

In [43]:
len(df[(df.difference == '2011') | (df.difference == '2012') | (df.difference == '2014') | (df.difference == '2015') | (df.difference == '2016')])

23

In [35]:
df[((df.difference == '2011') | (df.difference == '2012') | (df.difference == '2014') | (df.difference == '2015') | (df.difference == '2016') | (df.difference == '2017') | (df.difference == '2018') | (df.difference == '2019') | (df.difference == '2020')) & (df.difference != 'nan') & (df.hoogproblematisch == 'ja')]

idnr   gemeente (deelgemeente)        provincie      gewest  \
522   61010111  Antwerpen            NaN        Antwerpen  Vlaanderen   
1552  21010031    Beveren          Kallo  Oost-Vlaanderen  Vlaanderen   

                                  brugnaam probleem oorzaak werken_gepland  \
522   NOORDERLAAN-(STAD)-BRUG-1-INGANG-CNH      NaN     NaN            NaN   
1552     BEVERENTUNNEL WEGTUNNEL-VAN-KALLO      NaN     NaN            NaN   

     start_werken  ... beelden beheerder    beheer toestandcode  LPK  \
522           NaN  ...     NaN       AWV  volledig   Toestand 3  NaN   
1552          NaN  ...     NaN       AWV  volledig   Toestand 3  NaN   

     prev_year last_year difference hoogproblematisch lijst_sinds  
522        nan      2019       2019                ja         nan  
1552       nan      2011       2011                ja         nan  

[2 rows x 22 columns]

In [38]:
df[((df.difference == '2011') | (df.difference == '2012') | (df.difference == '2014') | (df.difference == '2015') | (df.difference == '2016')) & (df.difference != 'nan') & (df.toestandcode == 'Toestand 2')]

idnr   gemeente (deelgemeente)        provincie      gewest  \
678   01010007  Antwerpen        Berchem        Antwerpen  Vlaanderen   
1413  94500021    Beveren        Melsele  Oost-Vlaanderen  Vlaanderen   
1574  10021211    Beveren          Kallo  Oost-Vlaanderen  Vlaanderen   
1706  01004078       Gent            NaN  Oost-Vlaanderen  Vlaanderen   

                                               brugnaam probleem oorzaak  \
678   Keermuur aan de tunnel Gen. Lemastr. in Binnen...      NaN     NaN   
1413     BRUG-IN-N450-OVER-SPORENBUNDEL-EN-INDUSTRIEWEG      NaN     NaN   
1574                         BRUG-TOEGANG-STEENLANDLAAN      NaN     NaN   
1706                Keermuren aan brug Skaldenstraat R4      NaN     NaN   

     werken_gepland start_werken  ... beelden beheerder    beheer  \
678             NaN          NaN  ...     NaN       AWV  volledig   
1413            NaN          NaN  ...     NaN       AWV  volledig   
1574            NaN          NaN  ...     NaN       AWV  volledig   
1706            NaN          NaN  ...     NaN       AWV  volledig   

     toestandcode  LPK prev_year last_year difference hoogproblematisch  \
678    Toestand 2  NaN       nan      2016       2016               nee   
1413   Toestand 2  NaN       nan      2014       2014               nee   
1574   Toestand 2  NaN       nan      2016       2016               nee   
1706   Toestand 2  NaN       nan      2015       2015               nee   

     lijst_sinds  
678          nan  
1413         nan  
1574         nan  
1706         nan  

[4 rows x 22 columns]

## Welke problemen spelen bij deze bruggen?

In [557]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2748 entries, 0 to 2901
Data columns (total 22 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   idnr               2748 non-null   object
 1   gemeente           2748 non-null   object
 2   (deelgemeente)     1162 non-null   object
 3   provincie          2748 non-null   object
 4   gewest             2748 non-null   object
 5   brugnaam           2748 non-null   object
 6   probleem           131 non-null    object
 7   oorzaak            131 non-null    object
 8   werken_gepland     138 non-null    object
 9   start_werken       138 non-null    object
 10  bezocht            78 non-null     object
 11  opmerkingen        29 non-null     object
 12  beelden            20 non-null     object
 13  beheerder          2748 non-null   object
 14  beheer             2649 non-null   object
 15  toestandcode       2610 non-null   object
 16  LPK                42 non-null     object


In [612]:
df.oorzaak.value_counts()

onbekend       91
technisch      11
verouderd      10
groter plan     6
ongeval         4
betonrot        2
doorbuiging     2
corrosie        2
brokkelt        2
verzakking      1
Name: oorzaak, dtype: int64

In [616]:
df[df.oorzaak == 'technisch']

idnr  gemeente (deelgemeente)        provincie      gewest  \
1     00130041    Deurne         Deurne        Antwerpen  Vlaanderen   
2     00130031    Deurne         Deurne        Antwerpen  Vlaanderen   
2813       nan  Wijnegem            NaN        Antwerpen  Vlaanderen   
2816       nan  De Pinte            NaN  Oost-Vlaanderen  Vlaanderen   
2817       nan  De Pinte            NaN  Oost-Vlaanderen  Vlaanderen   
2822       nan    Kuurne            NaN  West-Vlaanderen  Vlaanderen   
2824       nan    Kuurne            NaN  West-Vlaanderen  Vlaanderen   
2826       nan   Schoten            NaN        Antwerpen  Vlaanderen   
2831       nan   Beersel            NaN   Vlaams Brabant  Vlaanderen   
2832       nan   Beersel            NaN   Vlaams Brabant  Vlaanderen   
2836       nan    Tienen            NaN   Vlaams Brabant  Vlaanderen   

                                               brugnaam  \
1                                Brug 2 - Sterckshoflei   
2                              Brug 1 - Cornelissenlaan   
2813                Brug op de N12 bij het Albertkanaal   
2816                                  Baron de Gieylaan   
2817                             Pont-Noord over de E17   
2822                            Kuurnebrug over de Leie   
2824  Fiets- en wandelbrug over de Heulebeek bij de ...   
2826                                      Hoogmolenbrug   
2831               Brug over de Zenne in de Zennestraat   
2832  Houten voetgangersbrug over de Kesterbeek in H...   
2836               Brug in de Biezenstraat over De Mene   

                                               probleem    oorzaak  \
1                  Jaarlijkse opvolging en waterpassing  technisch   
2     3-maandelijke inspectie, 6-maandelijkse waterp...  technisch   
2813      doorvaarthoogte is te laag en wordt opgehoogd  technisch   
2816                                  nieuwe asfaltlaag  technisch   
2817                                fietspaden voorzien  technisch   
2822                 brug ophogen ihkv Europees project  technisch   
2824  extra inzetten op verbindingswegen voor trage ...  technisch   
2826  In het kader van het verbeteren van de bereikb...  technisch   
2831  moet ervangen wegens toekomstige toename tonna...  technisch   
2832            brugdeik reinigen en plaatsen anti-slib  technisch   
2836       stabiliteitsproblemen, moet vernieuwd worden  technisch   

     werken_gepland        start_werken  ... beelden beheerder    beheer  \
1                Ja                2021  ...     NaN       AWV  volledig   
2                Ja                2022  ...     NaN       AWV  volledig   
2813             Ja                2023  ...     NaN  Gemeente       NaN   
2816             Ja            onbekend  ...     NaN  Gemeente       NaN   
2817             Ja            onbekend  ...     NaN  Gemeente       NaN   
2822             Ja                2021  ...     NaN  Gemeente       NaN   
2824             Ja                2022  ...     NaN  Gemeente       NaN   
2826             Ja               Bezig  ...     NaN  Gemeente       NaN   
2831             Ja  vermoedelijk >2024  ...     NaN  Gemeente       NaN   
2832             Ja                2020  ...     NaN  Gemeente       NaN   
2836             Ja            onbekend  ...     NaN  Gemeente       NaN   

     toestandcode                            LPK prev_year last_year  \
1      Toestand 4  2e prioriteit - te vernieuwen      2014      2017   
2      Toestand 4  1e prioriteit - te vernieuwen      2016      2019   
2813          NaN                            NaN       nan       nan   
2816          NaN                            NaN       nan       nan   
2817          NaN                            NaN       nan       nan   
2822          NaN                            NaN       nan       nan   
2824          NaN                            NaN       nan       nan   
2826          NaN                            NaN       nan       nan   
2831          NaN                 

In [615]:
df[df.oorzaak == 'verouderd']

idnr      gemeente (deelgemeente)        provincie      gewest  \
2814  nan      Mechelen            NaN        Antwerpen  Vlaanderen   
2815  nan        Voeren            NaN          Limburg  Vlaanderen   
2819  nan  Hamont-Achel            NaN          Limburg  Vlaanderen   
2821  nan     Merelbeke            NaN  Oost-Vlaanderen  Vlaanderen   
2823  nan        Kuurne            NaN  West-Vlaanderen  Vlaanderen   
2829  nan     Diksmuide            NaN  West-Vlaanderen  Vlaanderen   
2830  nan       Beersel            NaN   Vlaams Brabant  Vlaanderen   
2833  nan     Rotselaar            NaN   Vlaams Brabant  Vlaanderen   
2837  nan     Drogenbos            NaN   Vlaams Brabant  Vlaanderen   
2838  nan        Bertem            NaN   Vlaams Brabant  Vlaanderen   

                                               brugnaam  \
2814      Postzegelbrug (N1) over kanaal Leuven - Dijle   
2815                  Brug over "De Gulp" in Remersdaal   
2819                                      Brug Warmbeek   
2821      Brug over de Ringvaart en R4 bij Fraterstraat   
2823  Brug over de Heulebeek aan de monding van de Leie   
2829  Brug over de Grote Beverdijkvaart ten hoogte v...   
2830             Brug over de Zenne in de Dworpsestraat   
2833          Brug op de grens van Rotselaar en Tremelo   
2837      Brug in de Drie Fonteinenstraat over de Zenne   
2838           Brug over de dreef van kerk naar kasteel   

                                               probleem    oorzaak  \
2814  wordt afgebroken en komen twee bruggen voor in...  verouderd   
2815  is bouwvallig, afgesloten voor gemotoriseerd v...  verouderd   
2819                                           slijtage  verouderd   
2821                                           slijtage  verouderd   
2823                                          verouderd  verouderd   
2829                                          verouderd  verouderd   
2830                                          verouderd  verouderd   
2833                                          verouderd  verouderd   
2837  was lang niet in gebruik, helemaal afgebroken ...  verouderd   
2838   ouderdom, werken aan de stabiliteit, geen gevaar  verouderd   

     werken_gepland              start_werken  ... beelden beheerder beheer  \
2814             Ja                      2024  ...     NaN  Gemeente    NaN   
2815             Ja                      2020  ...     NaN  Gemeente    NaN   
2819             Ja                      2020  ...     NaN  Gemeente    NaN   
2821             Ja                      2021  ...     NaN  Gemeente    NaN   
2823             Ja                      2023  ...     NaN  Gemeente    NaN   
2829             Ja                      2012  ...     NaN  Gemeente    NaN   
2830             Ja                      2012  ...     NaN  Gemeente    NaN   
2833             Ja                 2021/2022  ...     NaN  Gemeente    NaN   
2837             Ja                     Bezig  ...     NaN  Gemeente    NaN   
2838             Ja  dit jaar of volgend jaar  ...     NaN  Gemeente    NaN   

     toestandcode  LPK prev_year last_year difference hoogproblematisch  \
2814          NaN  NaN       nan       nan        nan                ja   
2815          NaN  NaN       nan       nan        nan                ja   
2819          NaN  NaN       nan       nan        nan          onbekend   
2821          NaN  NaN       nan       nan        nan          onbekend   
2823          NaN  NaN       nan       nan        nan          onbekend   
2829          NaN  NaN       nan       nan        nan          onbekend   
2830          NaN  NaN       nan       nan        nan          onbekend   
2833          NaN  NaN       nan       nan        nan          onbekend   
2837          NaN  NaN       nan       nan        nan          onbekend   
2838          NaN  NaN       nan       nan        nan               nee   

     lijst_sinds  
2814         nan  
2815         nan  
2819         nan  
2821         nan  
2823       

In [617]:
df[df.oorzaak == 'groter plan']

idnr  gemeente (deelgemeente)        provincie      gewest  \
4   10000541  Zaventem       Zaventem   Vlaams Brabant  Vlaanderen   
5   10000542  Zaventem       Zaventem   Vlaams Brabant  Vlaanderen   
8   00030091  Zaventem       Zaventem   Vlaams Brabant  Vlaanderen   
13  00100451      Gent        Drongen  Oost-Vlaanderen  Vlaanderen   
17  00100461      Gent        Drongen  Oost-Vlaanderen  Vlaanderen   
18  00100441      Gent        Drongen  Oost-Vlaanderen  Vlaanderen   

                                      brugnaam  \
4                   Brug B16 Hector Hennaulaan   
5                Brug B16 - deel oprit naar R0   
8                           Viaduct B9 over R0   
13  Brug LXXIV over de A10 aan op- en afrit 13   
17                   Brug LXXV - Solvijnsdreef   
18                      Brug LXXIII in Drongen   

                                  probleem      oorzaak werken_gepland  \
4   Moet vernieuwd worden bij verbreden R0  groter plan             Ja   
5   Moet vernieuwd worden bij verbreden R0  groter plan             Ja   
8   Moet vernieuwd worden bij verbreden R0  groter plan             Ja   
13     Wordt vernieuwd met complex Drongen  groter plan             Ja   
17     Wordt vernieuwd met complex Drongen  groter plan             Ja   
18     Wordt vernieuwd met complex Drongen  groter plan             Ja   

   start_werken  ...        beelden beheerder    beheer toestandcode  \
4          2022  ...            NaN       AWV  volledig   Toestand 4   
5          2022  ...            NaN       AWV  volledig   Toestand 4   
8      onbekend  ...            NaN       AWV  volledig   Toestand 4   
13         2021  ...            Nee       AWV  volledig   Toestand 4   
17         2021  ...  Foto en video       AWV  volledig   Toestand 4   
18         2021  ...  Foto en video       AWV  volledig   Toestand 4   

                              LPK prev_year last_year difference  \
4   1e prioriteit - te vernieuwen      2015      2018          3   
5   1e prioriteit - te vernieuwen      2013      2018          5   
8   2e prioriteit - te vernieuwen      2014      2018          4   
13  2e prioriteit - te vernieuwen      2017      2020          3   
17  2e prioriteit - te vernieuwen      2016      2019          3   
18  2e prioriteit - te vernieuwen      2016      2019          3   

   hoogproblematisch lijst_sinds  
4                 ja        2016  
5                 ja        2007  
8                 ja        2009  
13                ja        1997  
17                ja        2001  
18                ja        2001  

[6 rows x 22 columns]

In [618]:
df[df.oorzaak == 'ongeval']

idnr   gemeente (deelgemeente)        provincie      gewest  \
9   00070131      Halle      Buizingen   Vlaams Brabant  Vlaanderen   
22  00170451  Roeselare        Rumbeke  West-Vlaanderen  Vlaanderen   
27  02570007   Zwevegem            NaN  West-Vlaanderen  Vlaanderen   
33  25900171       Gent        Drongen  Oost-Vlaanderen  Vlaanderen   

                                            brugnaam  \
9                           BRUG-1 over R0 in Drasop   
22                   BRUG-B50-N36-ROESELARE-KERKHOVE   
27  BRUG-AAN-LA-FLANDRE-OVER-NW-SLUIS-3-OUDE-SLUIS-6   
33                                 BEEKSTRAATBRUG-W3   

                                             probleem  oorzaak werken_gepland  \
9   Brug afgesloten voor gemotoriseerd verkeer na ...  ongeval             Ja   
22                                  Schade na ongeval  ongeval             Ja   
27     Beschadigd na aanrijding, werken in uitvoering  ongeval             Ja   
33   Beschadigd na aanrijding. Werken gepland in 2020  ongeval             Ja   

   start_werken  ... beelden         beheerder    beheer toestandcode  \
9      onbekend  ...     NaN               AWV  volledig   Toestand 4   
22     onbekend  ...     Nee               AWV  volledig   Toestand 4   
27         2020  ...     NaN  Vlaamse Waterweg  volledig   Toestand 4   
33         2020  ...     NaN  Vlaamse Waterweg  volledig   Toestand 4   

                              LPK prev_year last_year difference  \
9   1e prioriteit - te herstellen      2014      2018          4   
22  1e prioriteit - te herstellen      2013      2017          4   
27  1e prioriteit - te herstellen      2013      2018          5   
33  1e prioriteit - te herstellen      2015      2018          3   

   hoogproblematisch lijst_sinds  
9                 ja        2018  
22                ja        2018  
27                ja        2017  
33                ja        2018  

[4 rows x 22 columns]

Bruggen die verouderd zijn, staan eigenlijk altijd onder beheer van de gemeente. Bruggen die onderdeel zijn van een groter plan om gerenoveerd te worden, staan allemaal al langere tijd op de lijst en zijn dus hoogproblematische bruggen. Bruggen waar een ongeval de oorzaak is om te renoveren, zijn allemaal vrij recent op de lijst gekomen en zijn ook hoogproblematische bruggen.

## Zijn voor alle bruggen al werken gepland?

In [619]:
df.werken_gepland.value_counts()

Ja          87
onbekend    37
Nee         14
Name: werken_gepland, dtype: int64

In [621]:
df[df.werken_gepland == 'Nee']

idnr       gemeente                (deelgemeente)        provincie  \
0   00130101      Wommelgem                     Wommelgem        Antwerpen   
6   00040131       Overijse                      Overijse   Vlaams Brabant   
10  00120046     Grimbergen             Strombeek - Bever   Vlaams Brabant   
11  00120047     Grimbergen             Strombeek - Bever   Vlaams Brabant   
12  00100471           Gent  Sint-Denijs-Westrem - Afsnee  Oost-Vlaanderen   
16  00100681       Wetteren                       Westrem  Oost-Vlaanderen   
24  27000151     Nieuwpoort                    Nieuwpoort  West-Vlaanderen   
25  25700151       Kortrijk                           NaN  West-Vlaanderen   
26  25100191       Kortrijk                           NaN  West-Vlaanderen   
29  25900071      Merelbeke                     Merelbeke  Oost-Vlaanderen   
32  25520101  Sint-Laureins                           NaN  Oost-Vlaanderen   
34  25900161           Gent                       Drongen  Oost-Vlaanderen   
35  25100481           Gent                           NaN  Oost-Vlaanderen   
38  23000221          Halle                         Halle   Vlaams Brabant   

        gewest                                           brugnaam  \
0   Vlaanderen                         Brug B5 - Wijnegemsteenweg   
6   Vlaanderen  Brug B1 in weg van groot verkeer 305 bij op- e...   
10  Vlaanderen                                           BRUG-W20   
11  Vlaanderen                            BRUG-W21-IN-A12-OVER-R0   
12  Vlaanderen                Goedingebrug / Brug 12 over de Leie   
16  Vlaanderen                                Brug XC over de E40   
24  Vlaanderen                               LANGEBRUG-HN1-IN-N34   
25  Vlaanderen                             VOETBRUG-OVER-SLUIS-11   
26  Vlaanderen  Leiebrug in de Budastraat over de zijarm van d...   
29  Vlaanderen      PROF.-VANDEPITTEBRUG WEGBRUG-W12 TER-HANDBRUG   
32  Vlaanderen                             OOSTHOEKBRUG LIEVEBRUG   
34  Vlaanderen                             DRONGENBRUG-WEGBRUG-W4   
35  Vlaanderen                                      KROMMEWALBRUG   
38  Vlaanderen                                       Bospoortbrug   

                                             probleem      oorzaak  \
0   Vallende brokstukken. Afgesloten voor alle gem...     brokkelt   
6                                             Verzakt   verzakking   
10                                        Doorbuiging  doorbuiging   
11                                        Doorbuiging  doorbuiging   
12                              Moet vernieuwd worden     onbekend   
16                                   Wordt afgebroken     onbekend   
24                            Onder verhoogd toezicht     onbekend   
25                                       Studie bezig     onbekend   
26                                   Wordt afgebroken     onbekend   
29        Moet gerenoveerd worden, maar niet dringend     onbekend   
32   Er komt andere brug, alleen is deze brug erfgoed     onbekend   
34                            Onderzoek herstel bezig     onbekend   
35                                 Prijsvraag lopende     onbekend   
38                        Wordt op termijn afgebroken     onbekend   

   werken_gepland start_werken  ...        beelden         beheerder  \
0             Nee     onbekend  ...            NaN               AWV   
6             Nee     onbekend  ...            NaN               AWV   
10            Nee     onbekend  ...            NaN               AWV   
11            Nee     onbekend  ...            NaN               AWV   
12            Nee     onbekend  ...  Foto en video               AWV   
16            Nee     onbekend  ...  Foto en video               AWV   
24            Nee     onbekend  ...            NaN           AWV/VWW   
25            Nee     onbekend  ...  Foto en video  Vlaamse Waterweg   
26            Nee     onbekend  ...  Foto en video  Vlaamse Waterweg   
29            Nee     onbekend

In [624]:
df[df.beheerder == 'Gemeente'].werken_gepland.value_counts()

Ja    37
Name: werken_gepland, dtype: int64

Er zijn nog een heel aantal bruggen op de lijst van prioritaire kunstwerken waarvoor nog geen werken gepland zijn. Voor de bruggen in beheer van de gemeenten zijn wel altijd al werken gepland.

In [627]:
df[df.werken_gepland == 'Nee'].gemeente.value_counts()

Gent             3
Kortrijk         2
Grimbergen       2
Wommelgem        1
Sint-Laureins    1
Overijse         1
Wetteren         1
Halle            1
Merelbeke        1
Nieuwpoort       1
Name: gemeente, dtype: int64

Het gaat dan om drie bruggen in Gent, twee in Kortrijk, twee in Grimbergen, en nog in een aantal andere gemeenten.

In [47]:
len(df[(df.toestandcode == 'Toestand 2') | (df.toestandcode == 'Toestand 3') | (df.toestandcode == 'Toestand 4')])

1629